python-client/giskard/push/contribution.py

```diff
@@ -39,8 +44,8 @@ def create_contribution_push(model, ds, idrow, df):
                     return res
 
     if model.meta.model_type == SupportedModelTypes.REGRESSION and _existing_shap_values(ds):
-        shap_res = _contribution_push(model, ds, idrow)
-        slice_df = ds.slice(lambda df: df.loc[[idrow]], row_level=False)
+        shap_res = _contribution_push(model, ds, df)
+        slice_df = ds.slice(lambda row: row.equals(df.iloc[0]))
```

Comment:
I'm not sure why we need to do a row by row comparison like that, it seems suboptimal.